### DISCUSSION TITLE
Issue -> Discussion conversion

### DISCUSSION THREAD

**Discussion by ferdnyc** [andyholmes/gnome-shell-extension-gsconnect#1593] [Ideas]
(This is mostly for the project maintainers, though anyone else is welcome to participate if they're interested.)

In the process of renaming the "fixed in master" label to "fixed in main" (done, just now), I happened to notice the "Convert Issues" button next to the labels with issues attached to them.

![image](https://user-images.githubusercontent.com/538020/229943775-18f30322-cd32-44bc-931b-277e3e4a6242.png)

I had no idea what that _meant_, so I went searching the docs.

It [turns out](https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions#converting-issues-based-on-labels), now that we have Discussions enabled in the repo, we have the ability to use Issue labels to _convert_ them to discussions. 

Basically, we can create (or reuse) one or more labels, apply them to Issues that should be Discussions, and then choose Convert Issues. We'll be able to choose a destination Category, and all of the Issues with that label will be converted.

After that point, applying that label to _additional_ issues will convert them automatically. So, it basically empowers the label as a tool for redirecting Issues to Discussions.

Could be handy, if we feel there are Issues being created that should be Discussions instead. (I don't have a strong feel for how much of that already exists, since I never knew about the feature until just now. I'd have to go back through the existing Issues and evaluate.)

Sound useful? Potentially?

**Comment by ferdnyc** [comment]
cc: @andyholmes @daniellandau @sonnyp

**Comment by andyholmes** [comment]
I like the idea. I honestly don't go through the old issues anymore, because the majority are support requests, whimsical feature requests, or other non-actionable items.

Not that there's anything wrong with those, but it turns the issue tracker into a bit of a time-swamp.

**Comment by daniellandau** [comment]
Sounds good to me

**Reply by ferdnyc** [reply]
Now somebody just has to go through all the backlog. **... #NOTIT!** 🫡

(I've been slogging through the mess that is Crowdin, this weekend; I figure as time-sinks go, that one at least has the potential to benefit some users. ...Until, of course, I slam into the severe limits of my ability to make _any_ intelligent decisions on issues of other languages, which unsurprisingly didn't take more than a couple of hours. Now the _remaining_ issues (of which there are plenty), the universe does not exist where I'd be able to help in any way with those.)

**Reply by andyholmes** [reply]
> Now somebody just has to go through all the backlog. ... #NOTIT! 🫡

I usually only end up doing this when I'm avoiding something else :stuck_out_tongue_closed_eyes:. I'm not sure if "just users" can label issues, but I'm also not opposed to on-boarding some triagers if the opportunity presents itself.

**Reply by ferdnyc** [reply]
> I'm not sure if "just users" can label issues, 

Not to my knowledge, except for the ones the issue templates auto-apply on their behalf. (Which register as actions by the reporter; always found that confusing TBH.) 

But, as you say there's always the option to have triagers; the "Triage" repo role exists for a reason. 

Anyone with that role has the ability to "manage" issues and pull requests, which presumably means not only labeling but also closing. Possibly some other PR permissions as well. (Approving workflow runs? Assigning review requests?)

I suppose another approach would be to find a bot to apply labels on users' behalf. (One more complex than issue labeler, which loves to come along to issues I create with full labeling and slap the "triage" sticker on there anyway a second or two later.) Theoretically a command-driven bot ("&#x40;&#x200b;labelbot add UX", that kind of thing) could accept commands from anyone we want, or everyone.

...Yeah, that already feels like more effort than it's worth.

**Comment by ferdnyc** [comment]
WTH? ...So, I just tried to set this up, since there's one issue (so far) I've found that should really be a Q&A.

I created a new label (`discuss:Q&A`), applied it to that discussion, and went to set up conversion.

Despite the docs [_explicitly saying_](https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions#converting-issues-based-on-labels) (emphasis mine):

> ### Converting issues based on labels
> You can convert all issues with the same label to discussions in bulk. **Future issues with this label will also automatically convert to the discussion and category you configure.**

Here's what the Labels interface shows me:

![image](https://user-images.githubusercontent.com/538020/231963077-27ebebba-48e8-4635-86ea-9000209fbe39.png)

So now I don't know _what_ to think. Maybe the label has to be attached to at least TWO issues, to set up bulk autoconversion!? \*sigh\*

**Reply by ferdnyc** [reply]
Inquiry to the GitHub overlords: https://github.com/orgs/community/discussions/52785

**Reply by ferdnyc** [reply]
...Aaand, no response in 15 days. :disappointed:

**Reply by andyholmes** [reply]
Well, I mean it can't hurt to just press the big button. At least it would clean up the existing tracker, which is sort of out of control.

A lot of those could just be flagged as `support` and turned into discussions; there's virtually never a case where GSConnect or KDE Connect is at fault for device not connecting.

**Reply by ferdnyc** [reply]
Oh, I **did** convert the _one_ issue I tagged so far (not claiming I've looked very hard) — plus a few more I've done manually on a one-off basis. I'd just really like to be able to set up a tag or two that can streamline that process by simple assignment, if that's actually possible. (As their docs, unlike their UI, clearly seem to indicate that it _is or should be_.)

...And if it's _not_ possible, I'd like them to say that and **fix their docs**, because docs that lie are a special type of infuriating.
